Add example to show how the steady state emittance module works

SebastienJoly opened the pull request:
I have adapted the code snippets from the slides for the INC section meeting as an example. I print both the analytical steady state emittance as well as the calculated one. Additionally, I display some plots to see the transient behaviors of the emittance evolution. I need the example to include it in the documentation.

fsoubelet commented:
Few changes made:

- I have removed unnecessary lines
- I have split the example in two: one for coupling & default behavior and one for excitation & explicit starting values / constraints (in longitudinal)
- I have added comments explaining more about behavior, outputs etc
- I have shadowed the plotting code (with `#!end-doc-part`)
- I have left TODOs to include outputs in specific places

Regarding my last change, it is easier to wait until the convention change is enacted then run things instead of making a mess of branches everywhere, so I defer to later™️.

Required left:
- [x] Fix the TODOs when upstream merges are done for conventions
- [x] Still needs a test with our result values compared to the ones of ELEGANT.
- [ ] Optional: have results when switching Xsuite's twiss columns with the values of ELEGANT

SebastienJoly commented:
I have added the elegant test. I checked which relative tolerance we can use to compare Xsuite and elegant results. The emittances agree within 5%, and the IBS growth rates agree within 6%.
I did not compare the vertical emittances as they are already the byproduct of the horizontal ones. Similarly, the vertical IBS growth rate can differ by up 30% in a case with low coupling. However, it is not so important as the values are negligible compared to the horizontal and longitudinal IBS growth rates.
Please not that I cannot run the test using pytest due to the parallel PR which need to be pushed. As we say in french _wallah c'est le bordel_.

fsoubelet commented:
@SebastienJoly thanks. I don't see the `ibs_elegant.txt` file used, is it needed? I'd also rather see the new test in the same file, and binary format for data but I can handle this easily.

SebastienJoly commented:
I removed ibs_elegant.txt, it is not needed.
Actually you raise a valid point, I only use the last rows of the text file in the text. Would you rather see the values hardcoded in a dict in the test file? In that case we do not need to add an additional file nor to use pandas.

fsoubelet commented:
I would like to see whatever is most memory and time efficient :)

SebastienJoly commented:
I believe this is the second option with the hardcoded elegant results directly in the test. I have modified the test accordingly and removed the file ibs_evolution.txt from the test_data.
What do you think?

SebastienJoly commented:
I believe this is the second option with the hardcoded elegant results directly in the test. I have modified the test accordingly and removed the file ibs_evolution.txt from the test_data.
What do you think?

fsoubelet commented:
I have merged the test vs elegant with the rest of the equilibrium tests. I think this PR is alright now but I want to fix the remaining TODOs and rerun the tests once to be sure, which now requires the IBS convention changes to go through in the main `xfields` and that's in Gianni's hands now.

SebastienJoly commented:
Ok perfect!
Do you have any time estimate concerning the IBS convention change merge with xfields?

fsoubelet commented:
No :(

fsoubelet commented:
Good news: the convention changes have been merged and were published in `xfields 0.23.0`.

I have synced with upstream and merged the changes back into `feature/equilibrium_emittance` in #11. I'll get on the TODOs in this PR so we can merge and be clean.

fsoubelet commented:
Hi @SebastienJoly I have finished the coupling `constraint` example. When running the `excitation` constraint it reaches convergence in 2 steps only. Are you sure about the parameter values? I remember it was running longer in your presentation.

fsoubelet commented:
Figured it out on zoom. All good with the examples, only TODO not ticked is the optional run with ELEGANT's values replacing Xsuite's in the Twiss.

fsoubelet reviewed (approved):
Ran equilibrium tests which are all good. Happy to merge this so we have a clean branch for PR.
